ui/app/components/app/whats-new-popup/whats-new-popup.js

```diff
@@ -0,0 +1,190 @@
+import React, { useContext, useMemo, useRef, useState, useEffect } from 'react';
+import { useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { I18nContext } from '../../../contexts/i18n';
+import { useEqualityCheck } from '../../../hooks/useEqualityCheck';
+import Button from '../../ui/button';
+import Popover from '../../ui/popover';
+import { updateViewedNotifications } from '../../../store/actions';
+import { getTranslatedUINoficiations } from '../../../../../shared/notifications';
+import { getSortedNotificationsToShow } from '../../../selectors';
+
+function getActionFunctionById(id) {
+  const actionFunctions = {
+    2: () => {
+      global.platform.openTab({
+        url:
+          'https://survey.alchemer.com/s3/6173069/MetaMask-Extension-NPS-January-2021',
+      });
+    },
+    3: () => {
+      global.platform.openTab({
+        url: 'https://community.metamask.io/t/about-the-security-category/72',
+      });
+    },
+  };
+
+  return actionFunctions[id];
+}
+
+const renderFirstNotification = ({ notification, id, date, idRefMap }) => {
+  const actionFunction = getActionFunctionById(id);
+  return (
+    <div
+      className={classnames(
+        'whats-new-popup__notification whats-new-popup__first-notification',
+      )}
+      key="whats-new-popop-notificatiion-0"
+      ref={idRefMap[id]}
+    >
+      {notification.image && (
+        <img
+          className="whats-new-popup__notification-image"
+          src={notification.image.src}
+          height={notification.image.height}
+          width={notification.image.width}
+        />
+      )}
+      <div className="whats-new-popup__notification-title">
+        {notification.title}
+      </div>
+      <div className="whats-new-popup__description-and-date">
+        <div className="whats-new-popup__notification-description">
+          {notification.description}
+        </div>
+        <div className="whats-new-popup__notification-date">{date}</div>
+      </div>
+      {notification.actionText && (
+        <Button
+          type="secondary"
+          className="whats-new-popup__button"
+          rounded
+          onClick={actionFunction}
+        >
+          {notification.actionText}
+        </Button>
+      )}
+    </div>
+  );
+};
+
+const renderSubsequentNotification = ({
+  notification,
+  id,
+  date,
+  index,
+  idRefMap,
+}) => {
+  const actionFunction = getActionFunctionById(id);
+  return (
+    <div
+      className={classnames('whats-new-popup__notification')}
+      key={`whats-new-popop-notificatiion-${index}`}
```

Comment:
Nit: The `id` would make for a better key here, because it's guaranteed to be unique, whereas the index could change potentially (e.g. if we ever get dynamic notifications, or support sorting, etc.). Plus that would let us simplify these functions and get rid of the `index` parameter.